plugins/modules/route53_health_check.py

```diff
@@ -636,17 +636,16 @@ def main():
             # If health_check_name is a unique identifier
             if module.params.get('use_unique_names'):
                 existing_checks_with_name = get_existing_checks_with_name()
+                # Add 'Name' tag to add name to health check
+                if not tags:
+                    tags = {}
+                tags['Name'] = health_check_name
```

Comment:
If you do this, you need to make sure "purge_tags" is updated to False, if you don't, then leaving `tags` empty results in all non-Name tags being purged.

Reply:
for current code change `purge_tags=False` does not remove existing tags.
Added integration test to verify the same
https://github.com/ansible-collections/amazon.aws/pull/1253/files#diff-0d5db09c227a69a9880f5e7f16fde22576839c70c836a2836edbee1cee26989dR114-R146

Reply:
I think what @tremble is saying (feel free to correct me) is that because `purge_tags`by default is `true`, if the user has not set `tags`, what you are doing here is likely different than what they intend. With this logic, what will effectively happen is the following, even if they had not set `tags`:
```yaml
route53_health_check:
  ...
  tags:
    Name: whatever
  purge_tags: true
```
This means the only tag remaining after the task runs will be the `Name` tag.

Reply:
What we would want from a testing scenario is to have the task that sets `use_unique_names`, sets `purge_tags=true` (or just doesn't set it, since this is the default), and does *not* set the `tags` parameter. Then check to make sure that this does *not* remove the other non-`Name` tags.

Reply:
Fixed, added integration tests as well.
Now code behavior is as below
- tags={}, purge_tags=false do nothing
- tags={}, purge_tags=true remove existing tags except 'Name' tag
- tags=None, purge_tags=true do nothing